client/api.c

```diff
@@ -704,6 +707,49 @@ TDNFOpenHandle(
                   TDNF_CONF_GROUP);
     BAIL_ON_TDNF_ERROR(dwError);
 
+    if (!IsNullOrEmptyString(pTdnf->pArgs->pszInstallRoot) &&
+        strcmp(pTdnf->pArgs->pszInstallRoot, "/"))
+    {
+        int nIsDir = 0;
+
+        dwError = TDNFAllocateStringPrintf(&pszCacheDir, "%s/%s",
+                                           pTdnf->pArgs->pszInstallRoot,
+                                           pTdnf->pConf->pszCacheDir);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        TDNF_SAFE_FREE_MEMORY(pTdnf->pConf->pszCacheDir);
+        pTdnf->pConf->pszCacheDir = pszCacheDir;
+        pszCacheDir = NULL;
+
+        dwError = TDNFAllocateStringPrintf(&pszRepoDir, "%s/%s",
+                                           pTdnf->pArgs->pszInstallRoot,
+                                           pTdnf->pConf->pszRepoDir);
+        BAIL_ON_TDNF_ERROR(dwError);
+
+        dwError = TDNFIsDir(pszRepoDir, &nIsDir);
+        if (dwError == ERROR_TDNF_SYSTEM_BASE + ENOENT)
+        {
+            nIsDir = 0;
+            dwError = 0;
+        }
+        BAIL_ON_TDNF_ERROR(dwError);
+        if (nIsDir)
+        {
+            TDNF_SAFE_FREE_MEMORY(pTdnf->pConf->pszRepoDir);
+            pTdnf->pConf->pszRepoDir = pszRepoDir;
+            pszRepoDir = NULL;
```

Comment:
Why is it assigned to NULL here and in cleanup there is call to TDNF_SAFE_FREE_MEMORY(pszRepoDir)?  Is it done on purpose?